rmw_zenoh_cpp/src/detail/rmw_context_impl_s.hpp

```diff
@@ -17,20 +17,19 @@
 
 #include <zenoh.h>
 
-# include <cstddef>
+#include <cstddef>
 #include <memory>
-#include <mutex>
 #include <optional>
 #include <string>
-#include <unordered_map>
 
 #include "graph_cache.hpp"
-#include "guard_condition.hpp"
-#include "liveliness_utils.hpp"
 #include "rmw_node_data.hpp"
 
-#include "rcutils/types.h"
-#include "rmw/rmw.h"
+#include "rmw/ret_types.h"
+#include "rmw/types.h"
+
+// Forward declaration
+struct Data;
```

Comment:
```suggestion
class Data;
```

More importantly can we move this in `public` scope within `class rmw_context_impl_s`. Right now it's polluting the global namespace.

Reply:
Good call, fixed in 2384f6ce3e8a56f097cc7051a987c21e351f26fd.